builders/integrations/oracles/supra.md

```diff
@@ -0,0 +1,202 @@
+
+Check out our developer docs [here](https://supraoracles.com/docs/overview/).
+
+
+## Price Feeds {: #price-feeds }
+
+Our Pull model uses a combination of Web2 and Web3 methods to achieve ultra-low latency when sending data from Supra to the destination chain.  Web2 methods are used to retrieve data from Supra, while Web3 smart contracts are utilized for cryptographic verification.  
+Please refer to the below resources for a better understanding of our price feeds.
+[Data Feeds](https://supraoracles.com/docs/price-feeds/) - This explains how Supra calculates the S-Value for data feeds. 
+[Data Feeds Catalog](https://supraoracles.com/docs/price-feeds/data-feeds-index/) - This provides a list of data feeds currently offered by Supra.
+[Available Networks](https://supraoracles.com/docs/price-feeds/pull-model/networks/) - Available networks and Supra contract addresses.
+
+### Example Implementation
+
+## Step 1: 
+
+Create The S-Value Pull Interface to verify the price data received.
+Add the following code to the solidity smart contract that you wish to retrieve an S-Value.
+
+```
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+interface ISupraOraclePull {
+
+    //Verified price data
+    struct PriceData {
+        // List of pairs
+        uint256[] pairs;
+        // List of prices
+        // prices[i] is the price of pairs[i]
+        uint256[] prices;
+        // List of decimals
+        // decimals[i] is the decimals of pairs[i]
+        uint256[] decimals;
+    }
+
+
+function verifyOracleProof(bytes calldata _bytesproof) 
+    external 
+    returns (PriceData memory);
+}
+```  
+This creates the interface that you will later apply in order to verify and fetch S-Values from Supra's Pull Contract.
+
+## Step 2: 
+
+Configure The S-Value Feed Address
+
+To verify and fetch the S-Value from a Supra Pull smart contract, first find the S-Value Pull Contract address for the Moonbeam Network. When you have the address, create an instance of the ISupraOraclePull using the interface we previously defined:
+
+```
+// Mock contract which can consume oracle pull data
+contract MockOracleClient {
+    // The oracle contract
+    ISupraOraclePull internal oracle;
+
+    // Event emitted when a pair price is received
+    event PairPrice(uint256 pair, uint256 price, uint256 decimals);
+
+    constructor(address oracle_) {
+        oracle = ISupraOraclePull(oracle_);
+    }
+}
+```
+
+## Step 3: 
+
+Receive and Verify the S-Value
+
+Next, copy the following code to the smart contract to verify the price data received: 
+
+```
+// Get the price of a pair from oracle data received from supra pull model
+  
+    function GetPairPrice(bytes calldata _bytesProof, uint256 pair) external                 
```

Comment:
what do you need to pass into the ` _bytesProof` parameter? and for the `pair` parameter, is it the pair ID (found [here](https://supraoracles.com/docs/data-feeds/data-feeds-index))?

if it's not a straight-forward process to get the `_bytesProof`, we can direct users back to your docs. But at the very least, it would be helpful to have an example of what you can pass in for Moonbase Alpha (if possible)

Reply:
Hi @eshaben, apologies for the late reply!

The `_bytesProof` parameter is a byte-encoded value that contains the cryptographic proof for the price data. This proof ensures the authenticity and integrity of the price data being fetched from Supra's oracle.

As for the pair parameter, it indeed refers to the pair ID, which uniquely identifies the asset pair for which you are fetching the price data. 

Getting the `_bytesProof` requires you to be part of a cryptographic process where you request data from Supra's oracle, and the oracle responds with the data along with the proof (the _bytesProof). The smart contract then uses this proof to verify the authenticity of the data. It is pretty straightforward in our documentation.

Reply:
Thanks for clarifying!